File: javascript-closure-angular/API/Client/RequestRoutesArrivalSearch.js
goog.provide('API.Client.RequestRoutesArrivalSearch');

/**
 * @record
 */
API.Client.RequestRoutesArrivalSearch = function() {}

/**
 * @type {!string}
 * @export
 */
API.Client.RequestRoutesArrivalSearch.prototype.id;

/**
 * @type {!Array<!string>}
 * @export
 */
API.Client.RequestRoutesArrivalSearch.prototype.departureLocationIds;

/**
 * @type {!string}
 * @export
 */
API.Client.RequestRoutesArrivalSearch.prototype.arrivalLocationId;

/**
 * @type {!API.Client.RequestTransportation}
 * @export
 */
API.Client.RequestRoutesArrivalSearch.prototype.transportation;

/**
 * @type {!Date}
 * @export
 */
API.Client.RequestRoutesArrivalSearch.prototype.arrivalTime;

/**
 * @type {!Array<!API.Client.RequestRoutesProperty>}
 * @export
 */
API.Client.RequestRoutesArrivalSearch.prototype.properties;

/**
 * @type {!API.Client.RequestRangeFull}
 * @export
 */
API.Client.RequestRoutesArrivalSearch.prototype.range;

File: scala-play-server/app/model/RequestRoutesArrivalSearch.scala
package model

import play.api.libs.json._
import java.time.OffsetDateTime

/**
  * Represents the Swagger definition for RequestRoutesArrivalSearch.
  */
@javax.annotation.Generated(value = Array("org.openapitools.codegen.languages.ScalaPlayFrameworkServerCodegen"), date = "2020-07-14T11:29:17.777Z[Etc/UTC]")
case class RequestRoutesArrivalSearch(
  id: String,
  departureLocationIds: List[String],
  arrivalLocationId: String,
  transportation: RequestTransportation,
  arrivalTime: OffsetDateTime,
  properties: List[RequestRoutesProperty],
  range: Option[RequestRangeFull]
)

object RequestRoutesArrivalSearch {
  implicit lazy val requestRoutesArrivalSearchJsonFormat: Format[RequestRoutesArrivalSearch] = Json.format[RequestRoutesArrivalSearch]
}

